plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsReader.java

```diff
@@ -140,14 +144,22 @@ public static TableStatistics makeTableStatistics(
                 .map(column -> Maps.immutableEntry(column.fieldId(), column.type()))
                 .collect(toUnmodifiableMap(Map.Entry::getKey, Map.Entry::getValue));
 
+        Domain pathDomain = getPathDomain(effectivePredicate);
+        effectivePredicate = effectivePredicate.filter((icebergColumnHandle, domain) -> !icebergColumnHandle.equals(pathColumnHandle()));
```

Comment:
this should use `.filter((column, domain) -> !isMetadataColumnId(column.getId()));`

https://github.com/trinodb/trino/blob/037118255a08b45b7a061991cd22628ff3ce01d4/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java#L157

add a test for stats with a predicate on the `$file_modified_time` column